src/core/agent/services/identifierService.ts

```diff
@@ -303,6 +313,150 @@ class IdentifierService extends AgentService {
       );
     }
   }
+
+  async createMultisig(
+    creatorIdentifier: string,
+    contacts: KeriContact[],
+    meta: Pick<
+      IdentifierMetadataRecordProps,
+      "displayName" | "colors" | "theme"
+    >
+  ): Promise<string | undefined> {
+    const creatorMetadata = await this.getMetadataById(creatorIdentifier);
+    this.validIdentifierMetadata(creatorMetadata);
+    const creatorAid = (await this.agent.modules.signify.getIdentifierByName(
+      creatorMetadata.signifyName as string
+    )) as Aid;
+    const otherAids = [];
+    for (const contact of contacts) {
+      const aid = await this.agent.modules.signify.resolveOobi(contact.oobi);
+      otherAids.push({
+        state: aid.response,
+      } as Pick<Aid, "state">);
+    }
+    const result = await this.agent.modules.signify.createMultisig(
+      creatorAid,
+      otherAids
+    );
+    const multisigId = result.op.name.split(".")[1];
+    await this.createIdentifierMetadataRecord({
+      id: multisigId,
+      displayName: meta.displayName,
+      method: IdentifierType.KERI,
+      colors: meta.colors,
+      theme: meta.theme,
+      signifyName: utils.uuid(),
+      opName: result.op.name, //we save the opName here to sync the multisig's status later
+      isPending: result.op.done ? false : true, //this will be updated once the operation is done
+    });
+    return multisigId;
+  }
+  async hasJoinedMultisig(msgSaid: string): Promise<boolean> {
+    const notifications: MultiSigIcpNotification[] =
+      await this.agent.modules.signify.getNotificationsBySaid(msgSaid);
+    if (!notifications.length) {
+      return false;
+    }
+    const exn = notifications[0].exn;
+    const signifyName = exn.a.name;
+    try {
+      const multiSig = await this.agent.modules.signify.getIdentifierByName(
+        signifyName
+      );
+      if (multiSig) {
+        return true;
+      }
+    } catch (e) {
+      return false;
+    }
+    return false;
+  }
+
+  async joinMultisig(
+    notificationId: string,
+    meta: Pick<
+      IdentifierMetadataRecordProps,
+      "displayName" | "colors" | "theme"
+    >
+  ): Promise<string | undefined> {
+    const savedMultisigNotification = await this.agent.genericRecords.findById(
+      notificationId
+    );
+    if (!savedMultisigNotification) {
+      return;
```

Comment:
This should also be an error if it really should be there, but can we not just pass the notification directly to this function instead of the ID? It doesn't make sense to me to have to fetch it again, because I presume this is triggered when we receive the notification.